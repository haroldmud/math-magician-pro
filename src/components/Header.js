import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/Header.css';

function Header() {
  return (
    <header>
      <h1>Math Magician</h1>
      <nav>
        <Link className="menu-item" to="/">Home</Link>
        <Link className="menu-item" to="/Calculator">Calculator</Link>
        <Link className="menu-item" to="/Quote">Quote</Link>
      </nav>
    </header>
  );
}

export default Header;
