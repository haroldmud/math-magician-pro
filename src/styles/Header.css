header {
  position: relative;
}

header h1 {
  font-size: 3.5rem;
}

header nav {
  position: absolute;
  width: 15rem;
  top: 3rem;
  right: 8rem;
}

header nav a {
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  font-size: 20px;
}
