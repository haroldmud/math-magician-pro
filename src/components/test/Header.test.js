import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../Header';
import '@testing-library/jest-dom';

describe('Test of the header page', () => {
  it('test the page rendering', () => {
    const headerTest = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(headerTest).toMatchSnapshot();
  });
});
