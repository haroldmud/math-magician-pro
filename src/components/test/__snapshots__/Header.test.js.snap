// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the header page test the page rendering 1`] = `
<header>
  <h1>
    Math Magician
  </h1>
  <nav>
    <a
      className="menu-item"
      href="#/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="menu-item"
      href="#/Calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className="menu-item"
      href="#/Quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </nav>
</header>
`;
